### PR TITLE
MM-12424 Remove unused props from KeyboardLayout and simplify methods

### PR DESCRIPTION
**Pull request by hmhealey**
This is some extra work done along with https://github.com/mattermost/mattermost-mobile/pull/2201. Of the removed props:
- `statusBarHeight` wasn't actually used
- `theme` was only used to add a background colour which was already set everywhere except for the Thread screen which was attempting to set it through a `style` prop that didn't work previously
- `behavior` and `keyboardVerticalOffset` were being passed from the Thread screen, but they didn't do anything. They were left over from using RN's built-in KeyboardAvoidingView

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12424

#### Checklist
- Added or updated unit tests (required for all new features)

#### Device Information
This PR was tested on: iOS Simulator (iPhone 5s, iOS 12), Nexus 5 (Android 6.0)

### CODE REVIEW

File: app/components/layout/keyboard_layout/keyboard_layout.js [hunk 1]
```diff
@@ -41,52 +40,36 @@ export default class KeyboardLayout extends PureComponent {
...
-            return;
-        }
-
-        const {endCoordinates} = e;
-        const {height} = endCoordinates;
-
-        this.setState({bottom: height});
+    onKeyboardWillShow = (e) => {
+        this.setState({
+            keyboardHeight: e?.endCoordinates?.height || 0,
```
Comment (hmhealey) [inline app/components/layout/keyboard_layout/keyboard_layout.js]: The new version of RN also supports this operator. I forget the JS name for it, but it's called the safe navigation operator in Groovy. It is just a proposal right now and NOT PART OF JS at this point, so let me know what you think about using it.

If we wanted to support it in the other libraries, we'd need the Babel plugin that RN uses

File: app/components/layout/keyboard_layout/index.js [hunk 2]
```diff
@@ -1,19 +1,6 @@
...
-
-import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
-
-import {getStatusBarHeight} from 'app/selectors/device';
-
 import KeyboardLayout from './keyboard_layout';
 
-function mapStateToProps(state) {
-    return {
-        statusBarHeight: getStatusBarHeight(state),
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/layout/keyboard_layout/index.js]: Yeah probably does not even need to be in the store anymore

File: app/components/layout/keyboard_layout/keyboard_layout.js [hunk 3]
```diff
@@ -41,52 +40,36 @@ export default class KeyboardLayout extends PureComponent {
...
+        });
     };
 
     render() {
-        const {children, theme, ...otherProps} = this.props;
-        const style = getStyleFromTheme(theme);
+        const layoutStyle = [this.props.style, style.keyboardLayout];
```
Comment (lieut-data) [inline app/components/layout/keyboard_layout/keyboard_layout.js]: Is this an RN-specific thing, or can you stack style attributes as arrays in React too? Neat, regardless :)

Comment (hmhealey) [reply app/components/layout/keyboard_layout/keyboard_layout.js]: I haven't actually tried in regular React, but being able to pass multiple styles in an array in RN is great. The only thing here is that I'm not sure if this works if `this.props.style` itself is an array

**Pull request merged**